.github/workflows/sanitizers.yml

```diff
@@ -0,0 +1,43 @@
+# Syntax reference:
+# https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions
+
+name: Rust Sanitizers
+permissions: read-all
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    types: [opened, synchronize]
+
+
+jobs:
+  miri:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Install a nightly with miri
+      uses: actions-rs/toolchain@v1
+      with:
+          # Use a nightly that included miri
+          # https://rust-lang.github.io/rustup-components-history/
+          toolchain: nightly-2022-08-27
+          override: true
+          components: miri
```

Comment:
It seems like `actions-rs/toolchain` is unmaintained? Maybe use https://github.com/dtolnay/rust-toolchain instead?

**Edit:** nvm I see this is changed later to use rustup built into the image.